examples/viewer/viewer.cc

```diff
@@ -304,181 +385,214 @@ static bool LoadObjAndConvert(float bmin[3], float bmax[3],
-                assert(size_t(3 * nf1 + k) < attrib.normals.size());
-                assert(size_t(3 * nf2 + k) < attrib.normals.size());
-                n[0][k] = attrib.normals[3 * nf0 + k];
-                n[1][k] = attrib.normals[3 * nf1 + k];
-                n[2][k] = attrib.normals[3 * nf2 + k];
-              }
-            }
-          } else {
-            invalid_normal_index = true;
-          }
 
-          if (invalid_normal_index) {
-            // compute geometric normal
-            CalcNormal(n[0], v[0], v[1], v[2]);
-            n[1][0] = n[0][0];
-            n[1][1] = n[0][1];
-            n[1][2] = n[0][2];
-            n[2][0] = n[0][0];
-            n[2][1] = n[0][1];
-            n[2][2] = n[0][2];
-          }
-        }
-
-        for (int k = 0; k < 3; k++) {
-          buffer.push_back(v[k][0]);
-          buffer.push_back(v[k][1]);
-          buffer.push_back(v[k][2]);
-          buffer.push_back(n[k][0]);
-          buffer.push_back(n[k][1]);
-          buffer.push_back(n[k][2]);
-          // Combine normal and diffuse to get color.
-          float normal_factor = 0.2;
-          float diffuse_factor = 1 - normal_factor;
-          float c[3] = {n[k][0] * normal_factor + diffuse[0] * diffuse_factor,
-                        n[k][1] * normal_factor + diffuse[1] * diffuse_factor,
-                        n[k][2] * normal_factor + diffuse[2] * diffuse_factor};
-          float len2 = c[0] * c[0] + c[1] * c[1] + c[2] * c[2];
-          if (len2 > 0.0f) {
-            float len = sqrtf(len2);
-
-            c[0] /= len;
-            c[1] /= len;
-            c[2] /= len;
-          }
-          buffer.push_back(c[0] * 0.5 + 0.5);
-          buffer.push_back(c[1] * 0.5 + 0.5);
-          buffer.push_back(c[2] * 0.5 + 0.5);
-
-          buffer.push_back(tc[k][0]);
-          buffer.push_back(tc[k][1]);
-        }
-      }
-
-      o.vb_id = 0;
-      o.numTriangles = 0;
-
-      // OpenGL viewer does not support texturing with per-face material.
-      if (shapes[s].mesh.material_ids.size() > 0 &&
-          shapes[s].mesh.material_ids.size() > s) {
-        o.material_id = shapes[s].mesh.material_ids[0];  // use the material ID
-                                                         // of the first face.
-      } else {
-        o.material_id = materials.size() - 1;  // = ID for default material.
-      }
-      printf("shape[%d] material_id %d\n", int(s), int(o.material_id));
-
-      if (buffer.size() > 0) {
-        glGenBuffers(1, &o.vb_id);
-        glBindBuffer(GL_ARRAY_BUFFER, o.vb_id);
-        glBufferData(GL_ARRAY_BUFFER, buffer.size() * sizeof(float),
-                     &buffer.at(0), GL_STATIC_DRAW);
-        o.numTriangles = buffer.size() / (3 + 3 + 3 + 2) /
-                         3;  // 3:vtx, 3:normal, 3:col, 2:texcoord
-
-        printf("shape[%d] # of triangles = %d\n", static_cast<int>(s),
-               o.numTriangles);
-      }
+			// For each mesh in the shape
```

Comment:
New for loop: For each mesh in a shape.

Because of this new for loop, the indent has increased for all the code lines. That is one difference. The other difference is that shapes[s].mesh is replaced by mesh = shapes[s].meshes[meshIndex].